### PR TITLE
Fixes #9807: When in changes only mode, with no heartbeat, 'rudder agent run' outputs 'error: Rudder agent was interrupted during execution by a fatal error.'

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9807

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: share/lib/common.sh [hunk 1]
```diff
@@ -31,3 +31,13 @@ if [ -t 1 ]; then
...
 fi
+
+# Information extracted from the policies
+RUDDER_JSON="${RUDDER_VAR}/cfengine-community/inputs/rudder.json"
+
+if [ -f "${RUDDER_JSON}" ]
+then
+  RUDDER_REPORT_MODE=$(grep RUDDER_REPORT_MODE "${RUDDER_JSON}" | cut -d'"' -f4)
+  AGENT_RUN_INTERVAL=$(grep AGENT_RUN_INTERVAL "${RUDDER_JSON}" | cut -d'"' -f4)
+fi
```
Comment (ncharles) [inline share/lib/common.sh]: could we have a fallback json is not available ?

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Commit modified

File: share/lib/common.sh (hunk 1 above)
Comment (amousset) [inline share/lib/common.sh]: Yes, we can!

**Comment by amousset** [comment]
Commit modified

**Comment by ncharles** [comment]
:+1:

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**